rasa_audiocodes/audiocodes.py

```
@@ -153,19 +166,44 @@ def handle_start_conversation(self, body: Dict[Text, Any]) -> Dict[Text, Any]:
         cid = body["conversation"]
         if cid in self.conversations:
             sanic.exceptions.abort(500, "Conversation already exists")
+        logger.debug(f"(handle_start_conversation) --- New Conversation has arrived. Conversation: {cid}")
         self.conversations[cid] = AudiocodesInput.Conversation(cid)
-        return {
-            "activitiesURL": f"/webhooks/audiocodes/conversation/{cid}/activities",
-            "disconnectURL": f"/webhooks/audiocodes/conversation/{cid}/disconnect",
-            "refreshURL": f"/webhooks/audiocodes/conversation/{cid}/keepalive",
-            "expiresSeconds": self.keep_alive,
-        }
+        urls = self.get_basic_urls(cid)
+        if self.websocket_is_on:
+            urls.update({"websocketURL": f"/webhooks/audiocodes/conversation/{cid}/websocket"}) 
+        return urls
+
+    def get_basic_urls(self, cid):
+        return  {
+                    "activitiesURL": f"/webhooks/audiocodes/conversation/{cid}/activities",
+                    "disconnectURL": f"/webhooks/audiocodes/conversation/{cid}/disconnect",
+                    "refreshURL": f"/webhooks/audiocodes/conversation/{cid}/keepalive",
+                    "expiresSeconds": self.keep_alive,
+                }
 
     def blueprint(
         self, on_new_message: Callable[[UserMessage], Awaitable[Any]]
     ) -> Blueprint:
         ac_webhook = Blueprint("ac_webhook", __name__)
 
+        @ac_webhook.websocket('/conversation/<cid>/websocket')
+        async def new_client_connection(request, ws, cid: Text):
+            if self.websocket_is_on is False:
+                return
```

raise?

---

To raise ConnectionRefusedError instead of the return on line 192?

---

I will raise ConnectionRefusedError